src/main/java/com/minecolonies/coremod/items/ItemCactusDoor.java

```diff
@@ -0,0 +1,99 @@
+package com.minecolonies.coremod.items;
+
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.coremod.creativetab.ModCreativeTabs;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockDoor;
+import net.minecraft.block.SoundType;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public class ItemCactusDoor extends Item
+{
+    private final Block block;
+    //TODO REMOVE -  private final String name;
+
+    ItemCactusDoor(final Block block, final String name)
+    {
+        super();
+        this.block = block;
+        setRegistryName(name);
+        super.setUnlocalizedName(Constants.MOD_ID.toLowerCase() + "." + name);
+        this.setCreativeTab(ModCreativeTabs.MINECOLONIES);
+
+    }
+
+    @Override
+    public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
+    {
+        if (facing != EnumFacing.UP)
+        {
+            return EnumActionResult.FAIL;
+        }
+        else
+            {
+            final IBlockState iblockstate = worldIn.getBlockState(pos);
+            final Block blockIn = iblockstate.getBlock();
+            if (!blockIn.isReplaceable(worldIn, pos))
+            {
+                pos = pos.offset(facing);
+            }
+
+            final ItemStack itemstack = player.getHeldItem(hand);
+            if (player.canPlayerEdit(pos, facing, itemstack) && this.block.canPlaceBlockAt(worldIn, pos))
+            {
+                EnumFacing enumfacing = EnumFacing.fromAngle((double)player.rotationYaw);
+                final int i = enumfacing.getFrontOffsetX();
+                final int j = enumfacing.getFrontOffsetZ();
+                final boolean flag = (i < 0 && hitZ < 0.5F || i > 0 && hitZ > 0.5F || j < 0 && hitX > 0.5F || j > 0 && hitX < 0.5F);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (7) used in the expression (maximum allowed 6). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS864)